### PR TITLE
feat(ref-pattern): add new 'ref-pattern' rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
'ref-pattern' rule, which replaces the legacy
'incorrect_ref_pattern' rule.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [x] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

**Review by pyrooka — APPROVED** [review]
Looks good!

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: packages/validator/src/plugins/validation/2and3/semantic-validators/walker.js [hunk 1]
```diff
@@ -6,20 +6,18 @@
 
 // Assertation 2:
 // In specific areas of a spec, allowed $ref values are restricted.
+// [Removed]
```
Comment (padamstx) [inline packages/validator/src/plugins/validation/2and3/semantic-validators/walker.js:9]: @dpopp07 @pyrooka While looking through the code to identify built-in rules, I noticed that I didn't actually remove the logic for the "incorrect_ref_pattern" legacy rule, so updated the PR with those changes.

**Pull request merged**